contracts/src/ocr2/aggregator_proxy.cairo

```diff
@@ -0,0 +1,286 @@
+            aggregator.decimals()
+        }
+
+        fn type_and_version() -> felt252 {
+            let phase = _current_phase::read();
+            let aggregator = IAggregatorDispatcher { contract_address: phase.aggregator };
+            aggregator.type_and_version()
+        }
+    }
+
+    #[constructor]
+    fn constructor(owner: ContractAddress, address: ContractAddress) {
+        SimpleReadAccessController::initializer(owner);
+        _set_aggregator(address);
+    }
+
+    // --- Ownership ---
+
+    #[view]
+    fn owner() -> ContractAddress {
+        Ownable::owner()
+    }
+
+    #[view]
+    fn proposed_owner() -> ContractAddress {
+        Ownable::proposed_owner()
+    }
+
+    #[external]
+    fn transfer_ownership(new_owner: ContractAddress) {
+        Ownable::transfer_ownership(new_owner)
+    }
+
+    #[external]
+    fn accept_ownership() {
+        Ownable::accept_ownership()
+    }
+
+    #[external]
+    fn renounce_ownership() {
+        Ownable::renounce_ownership()
+    }
+
+    // -- SimpleReadAccessController --
+
+    #[external]
+    fn has_access(user: ContractAddress, data: Array<felt252>) -> bool {
+        SimpleReadAccessController::has_access(user, data)
+    }
+
+    #[external]
+    fn check_access(user: ContractAddress) {
+        SimpleReadAccessController::check_access(user)
+    }
+
+    //
+
+    #[external]
+    fn propose_aggregator(address: ContractAddress) {
+        Ownable::assert_only_owner();
+        assert(!address.is_zero(), 'proposed address is 0');
+        _proposed_aggregator::write(address);
+
+        let phase = _current_phase::read();
+        AggregatorProposed(phase.aggregator, address);
+    }
+
+    #[external]
+    fn confirm_aggregator(address: ContractAddress) {
+        Ownable::assert_only_owner();
+        assert(!address.is_zero(), 'confirm address is 0');
+        let phase = _current_phase::read();
+        let previous = phase.aggregator;
+
+        let proposed_aggregator = _proposed_aggregator::read();
+        assert(address == proposed_aggregator, 'does not match proposed address');
+        _proposed_aggregator::write(starknet::contract_address_const::<0>());
+        _set_aggregator(proposed_aggregator);
+
+        AggregatorConfirmed(previous, address);
+    }
+
+    #[view]
+    fn proposed_latest_round_data() -> Round {
+        _require_access();
+        let address = _proposed_aggregator::read();
+        // TODO: zero address check?
+        let aggregator = IAggregatorDispatcher { contract_address: address };
+        aggregator.latest_round_data()
+    }
+
+    #[view]
+    fn proposed_round_data(round_id: felt252) -> Round {
+        _require_access();
+        let address = _proposed_aggregator::read();
+        // TODO: zero address check?
+        let round_id128: u128 = round_id.try_into().unwrap();
+        let aggregator = IAggregatorDispatcher { contract_address: address };
+        aggregator.round_data(round_id128)
+    }
+
```

Comment:
We're missing the `round_data` and `latest_round_data` view functions definitions